main.js

```diff
@@ -0,0 +1,21 @@
+const tabitems=document.querySelectorAll('.tab-item');
+const tabcontentitems=document.querySelectorAll('.tab-content-item');
+//select tab content item
+function selectItem(e){
+    removeBorder();
+    removeshow()
+    //add border to current tab
+    this.classList.add('tab-border');
+    //grab content item from the dom
+    const tabcontentitem=document.querySelector(`#${this.id}-content`)
+    //add show class
+    tabcontentitem.classList.add('show')
+}
```

Comment:
If there is no element with an ID matching "this.id + '-content'", tab content item will be null, and calling classList.add('show') on a null value will result in an error, breaking the functionality of the tab switching.

To handle this potential issue, you should add error handling or validation to ensure that the corresponding content element exists before attempting to add the show class.  for example 

  // grab content item from the DOM
    const contentItemId = `#${this.id}-content`;
    const tabcontentitem = document.querySelector(contentItemId);

    if (tabcontentitem) {
        // add show class if the content item exists
        tabcontentitem.classList.add('show');
    } else {
        console.error(`Content item with ID ${contentItemId} not found.`);
    }